src/content-types/abstract.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpClient } from '../http';
+
+/**
+ * Options to configure a content-type manager.
+ */
+export interface ContentTypeManagerOptions {
+  /**
+   * The name of the resource this manager handles.
+   */
+  resource: string;
+
+  /**
+   * Optional path override for the resource.
+   *
+   * If not provided, the resource name is used to construct the path.
+   */
+  path?: string;
+}
+
+/**
+ * Abstract base class for managing content types.
+ */
+export abstract class AbstractContentTypeManager {
+  /**
+   * Configuration options for the content-type manager.
+   */
+  protected readonly _options: ContentTypeManagerOptions;
```

Comment:
Wasn't sure about either storing the options and exposing shorthands through getters or simply having the individual properties defined there instead of options, both work, opinions welcome :shrug:

Reply:
I think ideally there should be a way to save this, but I don't see the way it's done here interfering with adding that in the future. Worst case, we deprecate this method after we add a way to manually+permanently add custom methods

Reply:
> should be a way to save this

Not sure what you mean by that tbh, but in any case those properties are part of an internal API and aren't exposed to the developer, so we can modify that in the future as needed without deprecation

Reply:
Ahh, maybe I misinterpreted the "storing the options", I just meant that it would be nice to be able to actually define route paths permanently instead of just per-use